### PR TITLE
check whether options that require OOP are enabled.

### PR DESCRIPTION
**Pull request by heejaechang**
no code lens check yet since we don't have option yet. but as soon as I have one, we will check that as well.

### CODE REVIEW

**Comment by heejaechang** [comment]
@KevinH-MS @panopticoncentral @srivatsn @mavasani can you take a look?

File: src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostClientServiceFactory.RemoteHostClientService.cs [hunk 1]
```diff
@@ -187,6 +198,41 @@ private void OnConnectionChanged(object sender, bool connected)
...
+            private bool TestImpactEnabled()
+            {
+                // TODO: think about how to get rid of this code. since we don't want any code that require foreground
+                //       thread to run
+                AssertIsForeground();
+
+                var shellSettingsManager = new ShellSettingsManager(Shell.ServiceProvider.GlobalProvider);
+                var writeableSettingsStore = shellSettingsManager.GetWritableSettingsStore(SettingsScope.UserSettings);
+
+                if (writeableSettingsStore.PropertyExists(@"CodeAnalysis\TestImpact\IsGloballyEnabled", "Value"))
```
Comment (KevinH-MS) [inline src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostClientServiceFactory.RemoteHostClientService.cs]: nit: consider extract local string const

**Comment by KevinH-MS** [comment]
👍

File: src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostClientServiceFactory.RemoteHostClientService.cs (hunk 1 above)
Comment (heejaechang) [inline src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostClientServiceFactory.RemoteHostClientService.cs]: done

**Pull request merged**